### PR TITLE
Upgrades

### PR DESCRIPTION
**Pull request by elprans**
* Bump dependencies, upgrade to EdgeDB 2
* Port to the TS query builder
* Remove dependency on AWS in included Docker script

### CODE REVIEW

File: service/data/edgedb/agreements.ts [hunk 1]
```diff
@@ -9,359 +11,230 @@ import {
...
-      agreement.description,
-      agreement.creation_time,
-      agreement.versions.map(mapVersionEntity)
+    return await this.run(async (connection) =>
+      e
+        .select(e.Agreement, (agreement) => ({
+          id: true,
+          name: true,
+          description: true,
+          creationTime: agreement.creation_time,
```
Comment (scotttrinh) [inline service/data/edgedb/agreements.ts:27]: This is a cool pattern for rewriting snake case to camel case. Going to file this away.

File: service/domain/agreements.ts [hunk 2]
```diff
@@ -1,73 +1,27 @@
...
 export class AgreementVersion {
   id: string;
-  agreementId?: string;
+  agreementId?: string | null;
   current: boolean;
   draft: boolean;
   texts?: AgreementText[];
   creationTime: Date;
-
-  constructor(
```
Comment (scotttrinh) [inline service/domain/agreements.ts:28]: Any reason to keep these (`AgreementVersion`, `AgreementText`, `Agreement`) as classes now that they don't have a constructor?

File: service/domain/agreements.ts [hunk 3]
```diff
@@ -1,73 +1,27 @@
...
-    this.creationTime = creationTime;
-  }
+  description: string | null;
+  creationTime: Date | null;
 }
 
 export class AgreementVersion {
   id: string;
-  agreementId?: string;
+  agreementId?: string | null;
```
Comment (scotttrinh) [inline service/domain/agreements.ts:10]: Is this still optional or is it like the other optional -> null changes in this file?

```suggestion
  agreementId: string | null;
```

How about `texts`, are their contexts where it still might be optional?

File: service/domain/agreements.ts [hunk 4]
```diff
@@ -1,73 +1,27 @@
...
-  }
 }
 
 export class AgreementText {
   id: string;
   title: string;
   text: string;
   culture: string;
-  versionId: string;
+  versionId?: string | null;
```
Comment (scotttrinh) [inline service/domain/agreements.ts:22]: Same question here about nullish vs nullable

**Review by scotttrinh — APPROVED** [review]
Great! Nice to see a pretty one-to-one comparison in a pretty real-world application 👍 A couple of small questions, no blockers.

File: service/domain/agreements.ts (hunk 3 above)
Comment (elprans) [reply service/domain/agreements.ts:10]: Good catch.  Fixed.

File: service/domain/agreements.ts (hunk 2 above)
Comment (elprans) [reply service/domain/agreements.ts:28]: I'm not sure.  There are other method-less classes in the codebase.

File: service/domain/agreements.ts (hunk 3 above)
Comment (elprans) [reply service/domain/agreements.ts:10]: Actually, seems like the build fails without these being both nullable and nullish.

**Pull request merged**